src/ChunkMap.h

```diff
@@ -460,15 +468,25 @@ class cChunkMap
 		/** Returns true if there is an entity with the specified ID within this layer's chunks */
 		bool HasEntity(UInt32 a_EntityID);
 
+		/** Returns how many chunks this layer has. */
+		size_t GetNumChunks();
+
 	protected:
 
 		cChunkPtr m_Chunks[LAYER_SIZE * LAYER_SIZE];
 		int m_LayerX;
 		int m_LayerZ;
 		cChunkMap * m_Parent;
-		int m_NumChunksLoaded;
 
 		cAllocationPool<cChunkData::sChunkSection> & m_Pool;
+	private:
+		/** Increase the internal chunk counter. */
+		void IncreaseChunkCounter();
+
+		/** Decrease the internal chunk counter. */
+		void DecreaseChunkCounter();
+
+		size_t m_ChunkCounter;
```

Comment:
I've been working under the assumption that incremental counting is always to be preferred over iterating all chunks, whether it's mobs, statistics, RAM, or chunk count.

It didn't show up in profiles because it was not being used often before. Right now it's used whenever a chunk is unloaded to determine if we should remove the chunkLayer. It's also used by each world once every 2 seconds.